rest-api/tools/consent_upload_validation.py

```diff
@@ -0,0 +1,139 @@
+    except subprocess.CalledProcessError:
+      print 'Skipping participant {}: Directory does not exist.'.format(_id)
+
+    participant_files[_id] = output_list
+
+  return participant_files
+
+
+def _strip_path(f):
+  return os.path.basename(f)
+
+
+def _sanitize_versions(f):
+  return re.sub(r'__\d+', '', f)
+
+
+def remove_path_and_version_info(participant_files):
+  for _id, files in participant_files.items():
+    strip = map(_strip_path, files)
+    no_versions = map(_sanitize_versions, strip)
+    participant_files[_id] = no_versions
+
+
+def get_missing_file_info(participant_files):
+  for _id, files in participant_files.items():
+    files = set(files)
+    missing_files = required_files.difference(files)
+    participant_files[_id] = {'files_found': list(files), 'missing_files': list(missing_files)}
+    print 'Missing Files for participant {}: {}'.format(_id, list(missing_files))
+
+
+def write_to_csv(participant_files, descriptor):
+  missing_fields = ['pmi_id', 'missing files']
+  found_fields = ['pmi_id', 'files found']
+  missing_filename = 'missing_files_' + descriptor + '_' + str(datetime.date.today()) + '.csv'
+  existing_filename = 'existing_files_' + descriptor + '_' + str(datetime.date.today()) + '.csv'
+  print 'Creating csv files...'
+  with open(missing_filename, 'w') as missing:
+    with open(existing_filename, 'w') as found:
+      missing_writer = csv.writer(missing)
+      found_writer = csv.writer(found)
+      missing_writer.writerow(missing_fields)
+      found_writer.writerow(found_fields)
+
+      for _id, files in participant_files.items():
+        missing_files = files['missing_files']
+        existing_files = files['files_found']
+        missing_files.insert(0, _id)
+        existing_files.insert(0, _id)
+        if len(missing_files) > 1:
+          missing_writer.writerow(missing_files)
+        if len(existing_files) > 1:
+          found_writer.writerow(existing_files)
+
+  print 'Created files: {} | {}'.format(missing_filename, existing_filename)
+
+
+def main(args):
+  participant_dict = read_csv(args.input)
+  participant_ids = list(participant_dict.keys())
+
+  participant_files_base_bucket = get_bucket_file_info(participant_ids, base_bucket)
+  participant_files_awardee_bucket = get_bucket_file_info(participant_ids, args.bucket,
+                                                          participant_dict)
+
+  for files in [participant_files_base_bucket, participant_files_awardee_bucket]:
+    remove_path_and_version_info(files)
+    get_missing_file_info(files)
+
+  write_to_csv(participant_files_base_bucket, descriptor=base_bucket)
+  write_to_csv(participant_files_awardee_bucket, descriptor=args.bucket)
+
+
+if __name__ == '__main__':
+  configure_logging()
+  parser = get_parser()
+  parser.add_argument('--bucket', help='The source bucket to check against. i.e. aouXXX',
+                      required=True)
+  parser.add_argument('--input', help='Path to input csv', required=True)
```

Comment:
What are the expected contents of the input CSV and what does this represent? I'm guessing it's participants for which we want to check for signature uploads? (please add to this help string)

Reply:
done